src/uct/tcp/tcp_ep.c

```diff
@@ -418,6 +418,49 @@ static inline void uct_tcp_ep_ctx_move(uct_tcp_ep_ctx_t *to_ctx,
     memset(from_ctx, 0, sizeof(*from_ctx));
 }
 
+static ucs_status_t uct_tcp_ep_setup_keepalive(uct_tcp_ep_t *ep)
+{
+#if UCT_TCP_EP_KEEPALIVE
+    uct_tcp_iface_t *iface = ucs_derived_of(ep->super.super.iface,
+                                            uct_tcp_iface_t);
+    const int optval       = 1;
+    int idle_sec;
+    int intvl_sec;
+    ucs_status_t status;
+
+    if (!uct_tcp_iface_is_keepalive_enabled(iface)) {
```

Comment:
why need this separate function?

Reply:
originally it should be used in iface query function: in case if TCP may be configured to keepalive - allow err-handling, if not supported - disallow

Reply:
ok